test/integration/controller/admissionchecks/provisioning/provisioning_test.go

```diff
@@ -1958,3 +1960,121 @@ var _ = ginkgo.Describe("Provisioning", ginkgo.Ordered, ginkgo.ContinueOnFailure
 		})
 	})
 })
+
+var _ = ginkgo.Describe("Infinite Preemption Loop", func() {
+	var (
+		ctx context.Context
+		ns  *corev1.Namespace
+	)
+
+	ginkgo.JustBeforeEach(func() {
+		fwk.StartManager(ctx, cfg, managerSetup())
+	})
+
+	ginkgo.BeforeEach(func() {
+		ctx = context.Background()
+
+		// Create a namespace for the test
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "preemption-loop-test-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+	})
+
+	ginkgo.AfterEach(func() {
+		// Cleanup namespace
+		gomega.Expect(k8sClient.Delete(ctx, ns)).To(gomega.Succeed())
+		fwk.StopManager(ctx)
+	})
+
+	ginkgo.It("should reproduce the infinite preemption loop", func() {
+		// Step 1: Create ClusterQueues and Cohort
+		cohortName := "testing"
+
+		clusterQueueA := testing.MakeClusterQueue("guaranteed-tenant-a").
+			Cohort(cohortName).
+			ResourceGroup(*testing.MakeFlavorQuotas("default").
+				Resource("cpu", "150m", "150m").Obj()).
```

Comment:
150m, and unset the borrowing limit ([2nd parameter](https://github.com/kubernetes-sigs/kueue/blob/db584b66fde753dc052387d99005364287540dbd/pkg/util/testing/wrappers.go#L867))